DistanceScraper/Scrapers/BaseScraper.cs

```diff
@@ -184,36 +183,49 @@ public async Task RequestUserInfo(List<SteamID> steamIDs)
 				return;
 			}
 
+			Utils.WriteLine($"Requesting names for {steamIDs.Count} players");
 			IsRequestingUsers = new TaskCompletionSource<bool>();
-			steamIDs.ForEach(x => UsersToRequest.Enqueue(x));
-			var next = UsersToRequest.Dequeue();
-			Utils.WriteLine($"Requesting name for {next}");
-			RequestedSteamID = next;
-			Handlers.Friends.RequestFriendInfo(next, EClientPersonaStateFlag.PlayerName);
+			RequestedSteamIDs.AddRange(steamIDs);
+			foreach(var steamID in steamIDs)
+			{
+				if (Settings.Verbose)
+				{
+					Utils.WriteLine($"Requesting name for {steamID}");
+				}
+			}
+			Handlers.Friends.RequestFriendInfo(steamIDs, EClientPersonaStateFlag.PlayerName);
 
 			await IsRequestingUsers.Task;
 		}
 
 		public void OnRequestUserInfo(PersonaStateCallback callback)
 		{
+			if (Settings.Verbose)
+			{
+				Utils.WriteLine($"RequestedSteamIDs: {RequestedSteamIDs.Count}");
+			}
+			
 			// Skip callbacks from stuff we don't care about
-			if(callback.FriendID != RequestedSteamID)
+			if(!RequestedSteamIDs.Contains(callback.FriendID))
 			{
+				if (Settings.Verbose)
+				{
+					Utils.WriteLine($"Unexpected SteamID: {callback.FriendID}");
+				}
 				return;
 			}
-
-			Utils.WriteLine($"Name found for      {callback.FriendID}");
-			if (UsersToRequest.Count == 0)
+			RequestedSteamIDs.Remove(callback.FriendID);
+			if (Settings.Verbose)
 			{
-				//Utils.WriteLine("Queue empty");
-				IsRequestingUsers.SetResult(true);
+				Utils.WriteLine($"Name found for      {callback.FriendID}");
```

Comment:
The extra spacing here is so that this line and the requesting line will line up in logs.
`Requesting name for {steamID}`
`Name found for      {callback.FriendID}`